tmt/steps/provision/__init__.py

```diff
@@ -1124,34 +1125,33 @@ def store_log(
             log_name: Optional[str] = None) -> None:
         """
         Save log content to a file.
+
         :param log_path: a path to save into,could be a directory
-        or a file path.
+            or a file path.
         :param log_content: content of the log.
         :param log_name: name of the log, if not set, log_path
-        is supposed to be '/dev/null' or  a file path.
+            is supposed to be a file path.
         """
-        # if log_path is file path or /dev/null
-        if not log_path.is_dir() or str(log_path) == '/dev/null':
+        # if log_path is file path
+        if not log_path.is_dir():
             log_path.write_text(log_content)
         # log_path is a directory
         else:
             if log_name:
-                name_str = log_name
+                (log_path / log_name).write_text(log_content)
             else:
-                name_str = 'tmt-guestlog-' + \
-                    datetime.datetime.now(datetime.timezone.utc).strftime("%Y%m%d%H%M%S")
-                self.warn(f'log_name is not set, using file name:{name_str}')
-            (log_path / name_str).write_text(log_content)
+                raise tmt.utils.GeneralError('log_name is None.')
```

Comment:
Let's help the user a bit :) How about "Log path is a directory but log name is not defined."? Or something similar, to provide more context.